nixos/modules/system/boot/plymouth.nix

```diff
@@ -184,19 +191,24 @@ in
               exit 1
           fi
 
-          mkdir $out
-          cp -r ${themesEnv}/share/plymouth/themes/${cfg.theme} $out
+          mkdir -p $out/${cfg.theme}
+          cp -r ${themesEnv}/share/plymouth/themes/${cfg.theme}/* $out/${cfg.theme}
           # Copy more themes if the theme depends on others
-          for theme in $(grep -hRo '/etc/plymouth/themes/.*$' $out | xargs -n1 basename); do
+          for theme in $(grep -hRo '/share/plymouth/themes/.*$' $out | xargs -n1 basename); do
               if [[ -d "${themesEnv}/share/plymouth/themes/$theme" ]]; then
                   if [[ ! -d "$out/$theme" ]]; then
                     echo "Adding dependent theme: $theme"
-                    cp -r "${themesEnv}/share/plymouth/themes/$theme" $out
+                    mkdir -p "$out/$theme"
+                    cp -r "${themesEnv}/share/plymouth/themes/$theme"/* "$out/$theme"
                   fi
               else
                 echo "Missing theme dependency: $theme"
               fi
           done
+          # Fixup references
+          for theme in $out/*/*.plymouth; do
+            sed -i "s,/nix/.*/share/plymouth/themes,$out," "$theme"
```

Comment:
again, *technically* the store isn't necessarily in `/nix/store`. I also *kind of* wonder if we can allow these files to be in their original locations in initrd rather than patching their locations here with `sed`. If the list of files is dynamic, that's a bit tricky, but still doable with a derivation that creates symlinks to the specific files we need.

Reply:
Without this bad things happen (I didn't originally have it and just had it symlink in)... Themes expect to be able to depend on each other and do so via relative paths that assume they are installed side-by-side

I will adjust it to use `builtins.storeDir` here too though, because we might as well. I'm pretty sure other things in the plymouth module also expect `/nix` (which is why I didn't just use `storeDir` initially), but I'll have to go double check and fix them too then

Reply:
Oh I remember what the issue now. It's because of the logo/watermark. You need it in the `ImageDir` of the theme, but it's not distributed with Plymouth and we replace it with the `boot.plymouth.logo` option's value

Therefore the `ImageDir` of the Plymouth `spinner` theme needs to be patched regardless unless we want to mess around with weird derivation assumptions or impure locations